Guide check for contribution

edwinjosegeorge opened the pull request:
Enable the checks for code quality locally.

  1.  Add pre-commit to run git hooks for flake8, black and mypy.
  2.  Fix: async pytest requires pytest-asyncio library to handle coroutines.

edwinjosegeorge committed:
build(test): async in pytest

Async in pytest requires pytest-asyncio plugin to be active

edwinjosegeorge committed:
build: adding pre-commit

Add pre-commit to poetry dev group to run git hook. pre-commit>3.6 requires python>=3.9,
conflicting project dependency

edwinjosegeorge committed:
docs: add pre-commit as contribution guidelines

edwinjosegeorge committed:
Merge branch 'refs/heads/main' into contribution_guide

krrishdholakia commented on poetry.lock:
```diff
@@ -427,6 +427,17 @@ files = [
 pycparser = "*"
 
+[[package]]
+name = "cfgv"
```
what is this? @edwinjosegeorge

edwinjosegeorge replied:
These packages was install as the dependencies of `pre-commit` package under `dev` group. 

![image](https://github.com/BerriAI/litellm/assets/52248484/2b4170a4-41ee-4050-8531-e21c3018d382)

krrishdholakia commented:
hey @edwinjosegeorge we already have this with pre-commit hooks 

I don't think you need to run 

```
poetry run flake8
poetry run pytest .
```


pre-commit should take care of this - https://github.com/BerriAI/litellm/blob/9b7db7ea47156b74a42fac9c0af2cc634943ac8b/.pre-commit-config.yaml#L4

edwinjosegeorge commented:
Hi @krrishdholakia

Thanks for the quick response ❤️ .  Upon cloning the repository, I found that there was no mention of `pre-commit` as a guide but I found the config file. The config file does runs `black`, `flake8` and `mypy`, but does not run 'pytest'. Installing and configuring `pre-commit` will surely run these, but not the test files. Maybe we can tag `black`, `flake8` on commit hook and `mypy`, 'pytest' on push hook. 

I also found couple of libraries that is imported, but not used. Running linters `pylint` can catch a lot of inconsistency improving the code quality.

edwinjosegeorge committed:
build: pytest in pre-commit

Adding pytest into pre-commit config at pre-push stage only

edwinjosegeorge committed:
build: mypy in pre-commit

mypy is not available locally; Using remote mirror

edwinjosegeorge committed:
docs: Update on contribution dev guide

edwinjosegeorge committed:
build: skip pytest from pre-commit

Pytest takes considerable time to run to complete, possibly slowing development process for devs

edwinjosegeorge committed:
refactor: merging config files

edwinjosegeorge commented:
closing as outdated
